crates/oxc_linter/src/rules/typescript/no_this_alias.rs

```diff
@@ -0,0 +1,213 @@
+use oxc_ast::{
+    ast::{AssignmentTarget, BindingPatternKind, Expression, SimpleAssignmentTarget},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(no-this alias): Unexpected aliasing of 'this' to local variable.")]
+#[diagnostic(severity(error), help("Unexpected aliasing of 'this' to local variable."))]
+struct NoThisAliasDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "typescript-eslint(no-this alias): Unexpected aliasing of members of 'this' to local variables."
+)]
+#[diagnostic(severity(error), help("Unexpected aliasing of members of 'this' to local variables."))]
```

Comment:
This help is somewhat unhelpful, maybe copying over the doc?

> Assigning a variable to this instead of properly using arrow lambdas may be a symptom of pre-ES6 practices or not managing scope well.

Reply:
Copied and wrote a more descriptive message for the destructures.